### PR TITLE
test: fix `integration-ddl-test`

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/35020

we should commit the txn even if it is a read txn because if the `gc` checker uses the same session, it will have a txn start with the same ts and cannot see the newest change.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: ddl/ddl.go [hunk 1]
```diff
@@ -1163,6 +1163,10 @@ func GetHistoryJobByID(sess sessionctx.Context, id int64) (*model.Job, error) {
 	}
 	t := meta.NewMeta(txn)
 	job, err := t.GetHistoryDDLJob(id)
+	if err != nil {
+		return nil, errors.Trace(err)
```
Comment (hawkingrei) [inline ddl/ddl.go:1167]: 1. If here return err. Do we need to roll back or cancel this txn?

Comment (xiongjiwei) [reply ddl/ddl.go:1167]: good point!

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**